spec/defines/resource_mailhost_spec.rb

```diff
@@ -136,29 +178,152 @@
 
           it { is_expected.to contain_concat__fragment("#{title}-header") }
           it param[:title] do
-            lines = catalogue.resource('concat::fragment', "#{title}-header").send(:parameters)[:content].split("\n")
-            expect(lines & Array(param[:match])).to eq(Array(param[:match]))
-            Array(param[:notmatch]).each do |item|
-              is_expected.to contain_concat__fragment("#{title}-header").without_content(item)
+            matches = Array(param[:match])
+
+            if matches.all? { |m| m.is_a? Regexp }
+              matches.each { |item| is_expected.to contain_concat__fragment("#{title}-header").with_content(item) }
+            else
+              lines = catalogue.resource('concat::fragment', "#{title}-header").send(:parameters)[:content].split("\n")
+              expect(lines & Array(param[:match])).to eq(Array(param[:match]))
             end
           end
         end
       end
     end
 
-    describe 'mailhost template content (SSL enabled)' do
+    describe "mailhost template content for imap" do
```

Comment:
I think Rubocop will also flag this, but use single quotes if there are no interpolated variables.